public/app/features/dashboard/utils/getPanelMenu.ts

```diff
@@ -202,8 +206,36 @@ export function getPanelMenu(
     subMenu: inspectMenu,
   });
 
+  const createAlert = async () => {
+    const formValues = await panelToRuleFormValues(panel, dashboard);
+
+    const ruleFormUrl = urlUtil.renderUrl('/alerting/new', {
+      defaults: JSON.stringify(formValues),
+      returnTo: location.pathname + location.search,
+    });
+
+    locationService.push(ruleFormUrl);
+  };
+
+  const onCreateAlert = (event: React.MouseEvent) => {
+    event.preventDefault();
+    createAlert();
+    reportInteraction('dashboards_panelheader_menu', { item: 'create-alert' });
+  };
+
+  const getAlertingMenuAvailability = () => {
+    const { unifiedAlertingEnabled } = getConfig();
+    const hasRuleReadPermissions = contextSrv.hasPermission(getRulesPermissions(GRAFANA_RULES_SOURCE_NAME).read);
+    const hasRuleUpdatePermissions = contextSrv.hasPermission(getRulesPermissions(GRAFANA_RULES_SOURCE_NAME).update);
+    const isAlertingAvailableForRead = unifiedAlertingEnabled && hasRuleReadPermissions;
+
+    return { isAlertingAvailableForRead, hasRuleUpdatePermissions };
+  };
```

Comment:
should this be moved out to a standalone function? Maybe moved to a file under /features/alerting ?

looks a bit strange to define it as a variable here and then call it right away (kind of defeats the purpose of a function)

Reply:
yes, it makes totally sense 💯

Reply:
yeah, I defined a function as I was re-using it in the previous code. Now it does not make sense if we keep it here, but as you said, it makes more sense to move it to alerting folder.